lib/src/gregorian_calendar_duration.dart

```diff
@@ -0,0 +1,157 @@
+import 'calendar_duration.dart';
+import 'gregorian_calendar.dart';
+import 'gregorian_consts.dart';
+
+/// Represents a given duration which can be added/subtracted
+/// from a [GregorianCalendar] object
+class GregorianCalendarDuration implements CalendarDuration<GregorianCalendar>{
+
+  GregorianCalendarDuration({this.days = 0, this.weeks = 0, this.months = 0, this.years = 0});
+
+  /// Constructs a GregorianCalendarDuration where the values have been normalized
+  factory GregorianCalendarDuration.normalized({int years = 0, int months = 0, int weeks = 0, int days=0}){
+    // normalize values and then call the real constructor
+    if (days >= daysPerWeek){
+      weeks += days ~/ daysPerWeek;
+      days %= daysPerWeek;
+    }
+    else if (days <= -daysPerWeek){
+      weeks += days ~/ daysPerWeek;
+      days = -(days.abs() % daysPerWeek);
+    }
+    if (months >= monthsPerYear){
+      years += months ~/ monthsPerYear;
+      months %= monthsPerYear;
+    }
+    else if (months <= -monthsPerYear){
+      years += months ~/ monthsPerYear;
+      months = -(months.abs() % monthsPerYear);
+    }
+    return GregorianCalendarDuration(years: years, months: months, weeks: weeks, days: days);
+  }
+
+  factory GregorianCalendarDuration.fromDays(int days, GregorianCalendar referenceDate){
+    final newReferenceDate = referenceDate.addDays(days);
+    return new GregorianCalendarDuration.normalized(
+      years: newReferenceDate.year - referenceDate.year,
+      months: newReferenceDate.month - referenceDate.month,
+      days:  newReferenceDate.day - referenceDate.day
+    );
+  }
+
```

Comment:
nit: additional whitespace here.

Reply:
And elsewhere in this file.

Reply:
Corrected. Whitespace should now be consistent.